x-pack/plugins/ml/public/application/components/ml_inference/types.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IngestInferenceProcessor } from '@elastic/elasticsearch/lib/api/types';
+
+export enum AddInferencePipelineSteps {
```

Comment:
It's not set in stone but in most cases we didn't use `enum` for things like this but a pattern that uses an object with string values and additionally export a union type on top, something like:

```ts
export const ADD_INFERENCE_PIPELINE_STEPS {
    CONFIGURATION: 'configuration',
    ADVANCED: 'advanced',
    TEST: 'test',
    CREATE: 'create',
} as const;
export type AddInferencePipelineSteps = typeof ADD_INFERENCE_PIPELINE_STEPS[keyof typeof ADD_INFERENCE_PIPELINE_STEPS];
```

Reply:
Good to be consistent - updated in 64ba3ac4ec279eb78f5f8a97d01005991ae76219